### ISSUE TITLE
How to give `sns.cluster_map` a precomputed distance matrix?

### ISSUE THREAD

**Issue by jolespin** [mwaskom/seaborn#989]
Usually when I do dendrograms and heatmaps, I use a distance matrix and do a bunch of `SciPy` stuff.  I want to try out `Seaborn` but `Seaborn` wants my data in rectangular form (rows=samples, cols=attributes, not a distance matrix) . 

An example of this would be:
http://stackoverflow.com/questions/2982929/plotting-results-of-hierarchical-clustering-ontop-of-a-matrix-of-data-in-python
or this 
http://hms-dbmi.github.io/scw/figure/module-detection-3.png

Is there a method to  give `sns.clustermap` a precomputed distance matrix?

**Maybe there is something I can tweak to get a square matrix out after giving it a rectangular matrix or by adjusting the parameters so it can take a distance matrix instead of a rectangular matrix?**

Here's the usage:

    seaborn.clustermap¶
    seaborn.clustermap(data, pivot_kws=None, method='average', metric='euclidean',
     z_score=None, standard_scale=None, figsize=None, cbar_kws=None, row_cluster=True,
     col_cluster=True, row_linkage=None, col_linkage=None, row_colors=None,
     col_colors=None, mask=None, **kwargs)

My code below:

    from sklearn.datasets import load_iris
    iris = load_iris()
    X, y = iris.data, iris.target
    DF = pd.DataFrame(X, index = ["iris_%d" % (i) for i in range(X.shape[0])], columns = iris.feature_names)
[![enter image description here][1]][1]

I don't think my method is correct below because I'm giving it a precomputed distance matrix and NOT a rectangular data matrix as it requests.  There's no examples I can find of how to use a correlation/distance matrix with `cluster_map` but there is for https://stanford.edu/~mwaskom/software/seaborn/examples/network_correlations.html but the ordering is not clustered w/ the plain `sns.heatmap` func. 

    DF_corr = DF.T.corr()
    DF_dism = 1 - DF_corr
    sns.clustermap(DF_dism)
[![enter image description here][2]][2]

  [1]: http://i.stack.imgur.com/U1Jpe.png
  [2]: http://i.stack.imgur.com/xHlZR.png

**Comment by mwaskom** [comment]
Please don't ask the same question in multiple channels. I saw your post on stackoverflow. I still don't 100% understand exactly what you are trying to do, but I think the answer is just that `clustermap` does not work the way you are hoping. The distance matrix is computed to organize the rows and columns, but the heatmap shows the data values and not the distance matrix.

**Comment by jolespin** [comment]
Ok, I tried to make this question more specific and leading into a request feature for inputting custom distance matrices where the seaborn backend computes a dendrogram.

**Comment by jolespin** [comment]
You can give it a precomputed by `linkage = hc.linkage(sp.distance.squareform(DF_dism), method='average');
sns.clustermap(DF_dism, row_linkage=linkage, col_linkage=linkage)`
